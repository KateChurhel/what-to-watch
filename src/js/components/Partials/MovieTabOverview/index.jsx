// libraries
import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
// helpers
import { computeRatingLevel } from '../../../helpers/rating';

const MovieTabOverview = ({ voteAverage, voteCount, overview }) => {
  const ratingLevel = useMemo(() => computeRatingLevel(voteAverage), [voteAverage]);

  return (
    <>
      <div className="movie-rating">
        <div className="movie-rating__score">{voteAverage}</div>
        <p className="movie-rating__meta">
          <span className="movie-rating__level">{ratingLevel}</span>
          <span className="movie-rating__count">{`${voteCount} ratings`}</span>
        </p>
      </div>
      <p className="movie-card__text">{overview}</p>
    </>
  );
};

MovieTabOverview.defaultProps = {
  voteAverage: 0,
  voteCount: 0,
  overview: '',
};

MovieTabOverview.propTypes = {
  voteAverage: PropTypes.number,
  voteCount: PropTypes.number,
  overview: PropTypes.string,
};

export default MovieTabOverview;
