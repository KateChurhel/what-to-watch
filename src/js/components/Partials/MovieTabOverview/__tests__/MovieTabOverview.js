import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import MovieTabOverview from '../index';

const MOCK_DATA = {
  voteAverage: 7.3,
  voteCount: 10,
  overview: 'Test Overview',
};

describe('Movies Tab Overview: ', () => {
  it('correctly renders with data', () => {
    const tree = shallow(<MovieTabOverview {...MOCK_DATA} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });

  it('correctly renders without data', () => {
    const tree = shallow(<MovieTabOverview />);
    expect(toJSON(tree)).toMatchSnapshot();
  });
});
