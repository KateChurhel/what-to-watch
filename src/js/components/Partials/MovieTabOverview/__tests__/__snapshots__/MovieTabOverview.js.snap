// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies Tab Overview:  correctly renders with data 1`] = `
<Fragment>
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      7.3
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Good
      </span>
      <span
        className="movie-rating__count"
      >
        10 ratings
      </span>
    </p>
  </div>
  <p
    className="movie-card__text"
  >
    Test Overview
  </p>
</Fragment>
`;

exports[`Movies Tab Overview:  correctly renders without data 1`] = `
<Fragment>
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      0
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Bad
      </span>
      <span
        className="movie-rating__count"
      >
        0 ratings
      </span>
    </p>
  </div>
  <p
    className="movie-card__text"
  />
</Fragment>
`;
